File: extract_steps.py
import re

with open('stance_app/lib/data/content.dart', 'r') as f:
    content = f.read()

# Find the workouts list
workout_match = re.search(r'static final List<Workout> workouts = \[(.*?)\]\s*};', content, re.DOTALL)
if not workout_match:
    # Try without the closing brace if it's at the end of file
    workout_match = re.search(r'static final List<Workout> workouts = \[(.*)', content, re.DOTALL)

if workout_match:
    workouts_text = workout_match.group(1)
    # Find all ExerciseStep calls
    steps = re.findall(r'ExerciseStep\((.*?)\),', workouts_text, re.DOTALL)
    
    print(f"Found {len(steps)} steps")
    for i, step in enumerate(steps):
        # Simple extraction of name and duration
        name_match = re.search(r'name:\s*\'(.*?)\'', step)
        dur_match = re.search(r'duration:\s*(\d+)', step)
        drill_match = re.search(r'drillId:\s*\'(.*?)\'', step)
        
        name = name_match.group(1) if name_match else "Unknown"
        dur = dur_match.group(1) if dur_match else "0"
        drill = drill_match.group(1) if drill_match else "None"
        
        print(f"{i}|{name}|{dur}|{drill}")
else:
    print("Workouts list not found")

File: extract_steps.js
const fs = require('fs');

const content = fs.readFileSync('stance_app/lib/data/content.dart', 'utf8');
const workoutStart = content.indexOf('static final List<Workout> workouts = [');
if (workoutStart === -1) {
    console.log('Workouts list not found');
    process.exit(1);
}

const workoutsText = content.substring(workoutStart);
const stepRegex = /ExerciseStep\((.*?)\),/gs;
let match;
const steps = [];

while ((match = stepRegex.exec(workoutsText)) !== null) {
    const stepContent = match[1];
    const nameMatch = stepContent.match(/name:\s*['"](.*?)['"]/);
    const durMatch = stepContent.match(/duration:\s*(\d+)/);
    const drillMatch = stepContent.match(/drillId:\s*['"](.*?)['"]/);
    
    steps.push({
        name: nameMatch ? nameMatch[1] : 'Unknown',
        duration: durMatch ? parseInt(durMatch[1]) : 0,
        drillId: drillMatch ? drillMatch[1] : 'None'
    });
}

console.log(`Found ${steps.length} steps`);
steps.forEach((s, i) => {
    console.log(`${i}|${s.name}|${s.duration}|${s.drillId}`);
});
